src/sparseml/transformers/export.py

```diff
@@ -123,50 +133,50 @@ def export_transformer_to_onnx(
     :return: path to the exported ONNX file
     """
-    task = "-".join(task.lower().split("_"))
-    if task not in _TASK_TO_CONSTRUCTOR:
-        raise ValueError(
-            f"task {task} unsupported for export_transformer_to_onnx. Supported "
-            f"tasks include {list(_TASK_TO_CONSTRUCTOR.keys())}"
-        )
-    auto_model_constructor = _TASK_TO_CONSTRUCTOR[task]
+    task = task.replace("_", "-").replace(" ", "-")
 
-    if not os.path.isdir(model_path):
+    if not os.path.exists(model_path) or not os.path.isdir(model_path):
         raise ValueError(
             "model_path must be a directory that contains the trained transformer "
-            f"files. {model_path} is not a directory"
+            f"files. {model_path} is not a directory or does not exist"
         )
 
-    # load config and tokenizer
+    _LOGGER.info(f"Attempting onnx export for model at {model_path} for task {task}")
     config_args = {"finetuning_task": finetuning_task} if finetuning_task else {}
-    config = AutoConfig.from_pretrained(model_path, **config_args)
+    config = AutoConfig.from_pretrained(
+        model_path,
+        **config_args,
+    )
     tokenizer = AutoTokenizer.from_pretrained(
         model_path, model_max_length=sequence_length
     )
-
-    # load model
-    model = auto_model_constructor.from_pretrained(
-        model_path,
-        from_tf=False,
-        config=config,
+    model = _load_task_model(task, model_path, config)
+    _LOGGER.info(f"loaded model, config, and tokenizer from {model_path}")
+
+    trainer = Trainer(
+        model=model,
+        model_state_path=model_path,
+        recipe=None,
+        recipe_args=None,
+        teacher=None,
     )
+    applied = trainer.apply_manager(epoch=math.inf, checkpoint=None)
 
-    # apply recipe if exists before loading model weights
-    recipe_path = os.path.join(model_path, RECIPE_NAME)
-    if os.path.isfile(recipe_path):
-        ScheduledModifierManager.from_yaml(recipe_path).apply(model)
+    if not applied:
+        _LOGGER.warning(
+            f"No recipes were applied for {model_path}, "
+            "check to make sure recipe(s) are stored in the model_path"
+        )
     else:
-        _LOGGER.warning(f"recipe not found under {recipe_path}")
-
-    # load weights
-    load_kwargs = {} if torch.cuda.is_available() else {"map_location": "cpu"}
-    state_dict = torch.load(os.path.join(model_path, WEIGHTS_NAME), **load_kwargs)
-    model.load_state_dict(state_dict)
+        trainer.finalize_manager()
+        total_recipes = (1 if trainer.manager else 0) + len(trainer.arch_managers)
+        _LOGGER.info(f"Applied {total_recipes} total recipes the model at {model_path}")
 
     # create fake model input
     inputs = tokenizer(
         "", return_tensors="pt", padding=PaddingStrategy.MAX_LENGTH.value
     ).data  # Dict[Tensor]
+    _LOGGER.info(f"Created sample inputs for the ONNX export process: {inputs}")
```

Comment:
the inputs here are coming from a tokenized/padded empty string and will be just a large tensor of mostly 0s, ends up being more like log spam. could print out input shapes or names instead